lib/rubycritic/generators/json_report.rb

```diff
@@ -8,7 +8,10 @@ def initialize(analysed_modules)
       end
 
       def generate_report
-        print generator.render
+        FileUtils.mkdir_p(generator.file_directory)
```

Comment:
We can and should use [Pathname](https://ruby-doc.org/stdlib-2.1.0/libdoc/pathname/rdoc/Pathname.html#method-i-mkdir) here.